broker-core/src/main/java/io/zeebe/broker/subscription/message/state/MessageStartEventSubscriptionState.java

```diff
@@ -100,12 +109,18 @@ public void visitSubscriptionsByMessageName(
     subscriptionsColumnFamily.whileEqualPrefix(
         this.messageName,
         (key, value) -> {
-          visitor.visit(value);
+          final MessageStartEventSubscriptionRecord subscription =
+              new MessageStartEventSubscriptionRecord();
```

Comment:
Why do you create a new subscription instance?

Reply:
Each visit step overwrites subscriptionRecord. If a new instance is not created, visitors collecting subscriptions would contain references to the same instance. You can see that with out creating new instance, the test in MessageStartEventSubscriptionStateTest fails. Or may be I'm using UnpackedObjectValue wrongly.